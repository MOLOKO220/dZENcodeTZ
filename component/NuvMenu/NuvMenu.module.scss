.NuvMenu {
  max-width: 300px;
  width: 100vw;
  height: 100vh;
  padding: 80px 0 0 0;
  box-shadow: 5px 0px 20px rgba(114, 114, 114, 0.4);
  position: fixed;
  z-index: 800;
  left: -250px;
  transition: 300ms;
  background: white;
  > div {
    width: 100%;
    height: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
    position: relative;
    .NuvMenu__user_wrapp {
      height: 100px;
      width: 100px;
      margin: 30px 0;
      position: relative;
      > img {
        height: 100px;
        width: 100px;
        border-radius: 50%;
      }
      > div {
        background: white;
        height: 50px;
        width: 50px;
        border-radius: 50%;
        position: absolute;
        right: -20px;
        bottom: -20px;
        box-shadow: 1px 1px 13px rgba(0, 0, 0, 0.4);
        cursor: pointer;
        display: flex;
        align-items: center;
        justify-content: center;
      }
    }
    ul {
      display: flex;
      flex-direction: column;
      align-items: center;
      justify-content: center;
      font-family: "Roboto", sans-serif;
      > li {
        margin: 10px 0;
        transition: 300ms;
        &:hover {
          transform: scale(1.2);
        }
        > a {
          color: black;
        }
      }
    }
    .NuvMenu__arrow_btn {
      height: 30px;
      width: 30px;
      position: absolute;
      top: 10px;
      right: 10px;
      cursor: pointer;
      transition: 300ms;
    }
    .NuvMenu__arrow_btn_active {
      transform: rotate(0.5turn);
    }
  }
}
.NuvMenu__active {
  left: 0px;
}
