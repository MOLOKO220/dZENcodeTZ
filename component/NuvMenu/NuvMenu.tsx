import React, { useRef } from "react";
import styles from "./NuvMenu.module.scss";
import Link from "next/link";

export default function NuvMenu() {
  // hooks
  const NuvMenuWrapp = useRef<HTMLDivElement>(null);
  const arrowBtn = useRef<HTMLImageElement>(null);

  function showMenu() {
    NuvMenuWrapp.current?.classList.toggle(styles.NuvMenu__active);
    arrowBtn.current?.classList.toggle(styles.NuvMenu__arrow_btn_active);
  }

  return (
    <div className={styles.NuvMenu} ref={NuvMenuWrapp}>
      <div>
        <div className={styles.NuvMenu__user_wrapp}>
          <img src="user.jpg" alt="user" />
          <div /*onClick={}*/>
            <img src="gear.svg" alt="gear" />
          </div>
        </div>
        <ul>
          <li>
            <Link href="/">ПРИХОД</Link>
          </li>
          <li>
            <Link href="/Group">ГРУППЫ</Link>
          </li>
          <li>
            <Link href="#">ПРОДУКТЫ</Link>
          </li>
          <li>
            <Link href="#">ПОЛЬЗОВАТЕЛИ</Link>
          </li>
          <li>
            <Link href="#">НАСТРОЙКИ</Link>
          </li>
        </ul>
        <img
          className={styles.NuvMenu__arrow_btn}
          src="arrow.png"
          alt="arrow"
          ref={arrowBtn}
          onClick={showMenu}
        />
      </div>
    </div>
  );
}
