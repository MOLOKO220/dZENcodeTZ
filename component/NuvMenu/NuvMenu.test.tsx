import { render, screen } from "@testing-library/react";
import NuvMenu from "./NuvMenu";

describe("NuvMenu", () => {
  it(" render", () => {
    render(<NuvMenu />);
    expect(screen.getByText(/ПРИХОД/i)).toBeInTheDocument();
    expect(screen.getByText(/ГРУППЫ/i)).toBeInTheDocument();
    expect(screen.getByText(/ПРОДУКТЫ/i)).toBeInTheDocument();
    expect(screen.getByText(/ПОЛЬЗОВАТЕЛИ/i)).toBeInTheDocument();
    expect(screen.getByText(/НАСТРОЙКИ/i)).toBeInTheDocument();
  });

  it("snapshots", () => {
    const component = render(<NuvMenu />);
    expect(component).toMatchSnapshot();
  });
});
