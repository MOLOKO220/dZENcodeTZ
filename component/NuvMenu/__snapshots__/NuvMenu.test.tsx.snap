// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NuvMenu snapshots 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="NuvMenu"
      >
        <div>
          <div
            class="NuvMenu__user_wrapp"
          >
            <img
              alt="user"
              src="user.jpg"
            />
            <div>
              <img
                alt="gear"
                src="gear.svg"
              />
            </div>
          </div>
          <ul>
            <li>
              <a
                href="/"
              >
                ПРИХОД
              </a>
            </li>
            <li>
              <a
                href="/Group"
              >
                ГРУППЫ
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                ПРОДУКТЫ
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                ПОЛЬЗОВАТЕЛИ
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                НАСТРОЙКИ
              </a>
            </li>
          </ul>
          <img
            alt="arrow"
            class="NuvMenu__arrow_btn"
            src="arrow.png"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="NuvMenu"
    >
      <div>
        <div
          class="NuvMenu__user_wrapp"
        >
          <img
            alt="user"
            src="user.jpg"
          />
          <div>
            <img
              alt="gear"
              src="gear.svg"
            />
          </div>
        </div>
        <ul>
          <li>
            <a
              href="/"
            >
              ПРИХОД
            </a>
          </li>
          <li>
            <a
              href="/Group"
            >
              ГРУППЫ
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              ПРОДУКТЫ
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              ПОЛЬЗОВАТЕЛИ
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              НАСТРОЙКИ
            </a>
          </li>
        </ul>
        <img
          alt="arrow"
          class="NuvMenu__arrow_btn"
          src="arrow.png"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
